tests/core/reduction_test.py

```diff
@@ -225,6 +236,511 @@ def test_nanmean_single_dim(container):
+        sc.nanmedian(x, dim=['xx']), container(sc.scalar(5.0, unit='m'))
+    )
+    sc.testing.assert_identical(
+        x.nanmedian(dim=['xx']), container(sc.scalar(5.0, unit='m'))
+    )
+
+
+def test_nanmedian_single_dim(container):
+    x = container(
+        sc.array(dims=['xx', 'yy'], values=[[6, np.nan, 2], [np.nan, 5, 1]], unit='m')
+    )
+    sc.testing.assert_identical(
+        sc.nanmedian(x, dim='xx'),
+        container(sc.array(dims=['yy'], values=[6, 5, 1.5], unit='m')),
+    )
+    sc.testing.assert_identical(
+        x.nanmedian(dim='xx'),
+        container(sc.array(dims=['yy'], values=[6, 5, 1.5], unit='m')),
+    )
+    sc.testing.assert_identical(
+        sc.nanmedian(x, dim=['xx']),
+        container(sc.array(dims=['yy'], values=[6, 5, 1.5], unit='m')),
+    )
+    sc.testing.assert_identical(
+        x.nanmedian(dim=['xx']),
+        container(sc.array(dims=['yy'], values=[6, 5, 1.5], unit='m')),
+    )
+
+    sc.testing.assert_identical(
+        sc.nanmedian(x, dim='yy'),
+        container(sc.array(dims=['xx'], values=[4.0, 3.0], unit='m')),
+    )
+    sc.testing.assert_identical(
+        x.nanmedian(dim='yy'),
+        container(sc.array(dims=['xx'], values=[4.0, 3.0], unit='m')),
+    )
+    sc.testing.assert_identical(
+        sc.nanmedian(x, dim=['yy']),
+        container(sc.array(dims=['xx'], values=[4.0, 3.0], unit='m')),
+    )
+    sc.testing.assert_identical(
+        x.nanmedian(dim=['yy']),
+        container(sc.array(dims=['xx'], values=[4.0, 3.0], unit='m')),
+    )
+
+
+def test_var(container):
+    x = container(sc.array(dims=['xx', 'yy'], values=[[2, 5, 3], [2, 2, 4]], unit='m'))
+    # Yes, using identical with floats.
+    # It works and allclose doesn't support datasets and data groups.
+    sc.testing.assert_identical(
+        sc.var(x, ddof=0), container(sc.scalar(4 / 3, unit='m**2'))
+    )
+    sc.testing.assert_identical(x.var(ddof=0), container(sc.scalar(4 / 3, unit='m**2')))
+    sc.testing.assert_identical(
+        sc.var(x, ddof=0, dim=['xx', 'yy']), container(sc.scalar(4 / 3, unit='m**2'))
+    )
+    sc.testing.assert_identical(
+        x.var(ddof=0, dim=['xx', 'yy']), container(sc.scalar(4 / 3, unit='m**2'))
+    )
+
+    sc.testing.assert_identical(
+        sc.var(x, ddof=1), container(sc.scalar(1.6, unit='m**2'))
+    )
+    sc.testing.assert_identical(x.var(ddof=1), container(sc.scalar(1.6, unit='m**2')))
+    sc.testing.assert_identical(
+        sc.var(x, ddof=1, dim=['xx', 'yy']), container(sc.scalar(1.6, unit='m**2'))
+    )
+    sc.testing.assert_identical(
+        x.var(ddof=1, dim=['xx', 'yy']), container(sc.scalar(1.6, unit='m**2'))
+    )
+
+
+def test_var_single_dim(container):
+    x = container(sc.array(dims=['xx', 'yy'], values=[[2, 5, 3], [2, 2, 4]], unit='m'))
+    # Yes, using identical with floats.
+    # It works and allclose doesn't support datasets and data groups.
+    sc.testing.assert_identical(
+        sc.var(x, 'xx', ddof=0),
```

Comment:
I suggest to add separate tests to ensure that `ddof` is required, to prevent changes by a future developer not aware of our intentions.